### PR TITLE
WINDUP-842 removed breadcrumbs and add Application List into the top …

### PR DESCRIPTION
**Pull request by mareknovotny**
…navigation,

changed home icon to list for Java Application Overview and Migration issues Reports

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/647/

File: reporting/api/src/main/java/org/jboss/windup/reporting/model/ApplicationReportModel.java [hunk 1]
```diff
@@ -78,6 +79,22 @@
...
     @Property(DISPLAY_IN_APPLICATION_LIST)
     void setDisplayInApplicationList(Boolean displayInApplicationList);
+    
+
+    /**
+     * Indicates whether the Report is index of All Application 
+     * @return
+     */
+    @Property(GLOBAL_APPLICATION_LIST_REPORT)
+    Boolean getGlobalApplicationListReport();
```
Comment (jsight) [inline reporting/api/src/main/java/org/jboss/windup/reporting/model/ApplicationReportModel.java]: Is this a leftover from a previous idea? It doesn't really look like the value of this property is read anywhere?

Comment (jsight) [inline reporting/api/src/main/java/org/jboss/windup/reporting/model/ApplicationReportModel.java]: Oh, disregard that comment, I get it now. I had misread it before.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1273/

**Comment by jsight** [comment]
This seems to break in the case of multiple applications. The global navbar ends up having links to one of the two without it being clear which one it will be.

**Comment by mareknovotny** [comment]
@jsight how it breaks the multiapp navigation?
I tried specifically that see images for index and their both application overview https://owncloud-mnovotny.rhcloud.com/index.php/s/LFTaPy6NDIehXsh

**Comment by jsight** [comment]
@mareknovotny I'll try it again... maybe my build was bad somehow.

**Comment by jsight** [comment]
@mareknovotny - I tried it again and the application list still ends up with a navbar from one of the other applications. Basically this is random luck as all of the navbars now has a link to index.ftl with this PR. So when it tries to find the navbar in the index.ftl template, it will find one or the other. It's just luck as to whether it gets the right or the wrong one.

**Comment by mareknovotny** [comment]
@jsight could you paste/share for me somewhere your resulted reports to get an idea what is broken as I still can't see that.

**Comment by jsight** [comment]
@mareknovotny http://imgur.com/jk6XQ3a

**Comment by mareknovotny** [comment]
hmm, that is interesting, I will try to reproduce it, as I couldn't get that mixing of Application List report and navigation links from lower level of application reports, thanks!

**Comment by jsight** [comment]
@mareknovotny Perhaps rebase onto the latest master? I think it should be deterministic instead of random as to which one you end up with.

The code is here: https://github.com/windup/windup/blob/master/reporting/api/src/main/java/org/jboss/windup/reporting/model/ApplicationReportModel.java#L38

**Comment by mareknovotny** [comment]
oh, you changed ApplicationReportModel meanwhile, then it explains my confusion ;), I will rebase it and fix it tomorrow, thanks for explanation.

**Comment by jsight** [comment]
Yeah, although it was ambiguous before. This change just removes the ambiguity.

**Comment by mareknovotny** [comment]
ok, I took another more simplistic approach to get Application List into navbar. Added a variable to index.ftl and in navbar just test it and then render the index link if the variable doesn't exists. This is the same way as it was rendered in breadcrumbs.ftl as it was included in all templates except index.ftl. 

This PR also changes icons and text titles for WINDUP-838

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/664/<h2>Build result: FAILURE</span></h2>[...truncated 47749 lines...][ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[ERROR] [ERROR] After correcting the problems, you can resume the build with the command[ERROR]   mvn <goals> -rf :windup-victims[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/pom.xml to org.jboss.windup.jenkins/windup-core-dependents-aggregate-pom/2.3.0-SNAPSHOT/windup-core-dependents-aggregate-pom-2.3.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/proprietary-javaee-servlet/rules-java/pom.xml to org.jboss.windup.quickstarts/windup-proprietary-javaee-servlet-rules-java/2.5.0-SNAPSHOT/windup-proprietary-javaee-servlet-rules-java-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/proprietary-javaee-servlet/rules-java/target/windup-proprietary-javaee-servlet-rules-java-2.5.0-SNAPSHOT.jar to org.jboss.windup.quickstarts/windup-proprietary-javaee-servlet-rules-java/2.5.0-SNAPSHOT/windup-proprietary-javaee-servlet-rules-java-2.5.0-SNAPSHOT.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/proprietary-javaee-servlet/rules-java/target/windup-proprietary-javaee-servlet-rules-java-2.5.0-SNAPSHOT-forge-addon.jar to org.jboss.windup.quickstarts/windup-proprietary-javaee-servlet-rules-java/2.5.0-SNAPSHOT/windup-proprietary-javaee-servlet-rules-java-2.5.0-SNAPSHOT-forge-addon.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-rulesets/pom.xml to org.jboss.windup.rules/windup-rulesets/2.5.0-SNAPSHOT/windup-rulesets-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-rulesets/target/windup-rulesets-2.5.0-SNAPSHOT.jar to org.jboss.windup.rules/windup-rulesets/2.5.0-SNAPSHOT/windup-rulesets-2.5.0-SNAPSHOT.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-rulesets/target/windup-rulesets-2.5.0-SNAPSHOT-sources.jar to org.jboss.windup.rules/windup-rulesets/2.5.0-SNAPSHOT/windup-rulesets-2.5.0-SNAPSHOT-sources.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/proprietary-javaee-servlet/pom.xml to org.jboss.windup.quickstarts/windup-proprietary-javaee-servlet/2.5.0-SNAPSHOT/windup-proprietary-javaee-servlet-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/pom.xml to org.jboss.windup.quickstarts/windup-quickstarts-root/2.5.0-SNAPSHOT/windup-quickstarts-root-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/ejb-beanutils-async/rules-java/pom.xml to org.jboss.windup.quickstarts/windup-ejb-beanutils-async-rules-java/2.5.0-SNAPSHOT/windup-ejb-beanutils-async-rules-java-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/ejb-beanutils-async/rules-java/target/windup-ejb-beanutils-async-rules-java-2.5.0-SNAPSHOT.jar to org.jboss.windup.quickstarts/windup-ejb-beanutils-async-rules-java/2.5.0-SNAPSHOT/windup-ejb-beanutils-async-rules-java-2.5.0-SNAPSHOT.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/ejb-beanutils-async/rules-java/target/windup-ejb-beanutils-async-rules-java-2.5.0-SNAPSHOT-forge-addon.jar to org.jboss.windup.quickstarts/windup-ejb-beanutils-async-rules-java/2.5.0-SNAPSHOT/windup-ejb-beanutils-async-rules-java-2.5.0-SNAPSHOT-forge-addon.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/ejb-beanutils-async/pom.xml to org.jboss.windup.quickstarts/windup-ejb-beanutils-async/2.5.0-SNAPSHOT/windup-ejb-beanutils-async-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/victi.ms/pom.xml to org.jboss.windup.quickstarts/windup-victims/2.5.0-SNAPSHOT/windup-victims-2.5.0-SNAPSHOT.pomchannel stoppedArchiving artifactsIRC notifier plugin: Sending notification to: #windupSetting status of 29817474cf5353e2b0ef00198111a930f6b9c0d3 to FAILURE with url http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-core-PR-dependents-check/664/ and message: 'Build finished. 3 tests run, 0 skipped, 0 failed.'

**Comment by mareknovotny** [comment]
@windupgithubbot1 retest

**Comment by mareknovotny** [comment]
one thing is remaining which I overlooked, global migration issues report get twice App list

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1291/

File: reporting/api/src/main/java/org/jboss/windup/reporting/model/ApplicationReportIndexModel.java [hunk 2]
```diff
@@ -71,7 +71,8 @@
             List<ApplicationReportModel> reports = new ArrayList<>();
             for (ApplicationReportModel m : getApplicationReportModels())
             {
-                reports.add(m);
+                if (!m.getReportName().equals("Application List"))
```
Comment (jsight) [inline reporting/api/src/main/java/org/jboss/windup/reporting/model/ApplicationReportIndexModel.java]: Why is this check necessary?

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1292/

File: reporting/api/src/main/java/org/jboss/windup/reporting/model/ApplicationReportIndexModel.java (hunk 2 above)
Comment (mareknovotny) [inline reporting/api/src/main/java/org/jboss/windup/reporting/model/ApplicationReportIndexModel.java]: I added static link to index.html and I didn't get why Application List index report is there for global level reports so I filter result from getApplicationReportModelsSortedByPriority which is called in navBar.ftl (https://github.com/windup/windup/blob/master/reporting/impl/src/main/resources/reports/templates/include/navbar.ftl#L3).

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/665/

**Pull request merged**